scripts/dev_cli.sh

```diff
@@ -0,0 +1,315 @@
+    echo "        --integration        Run the integration tests."
+    echo "        --all                Run all tests."
+    echo ""
+    echo "    build-container [--type]"
+    echo "        Build the Cloud Hypervisor container."
+    echo "        --dev                Build dev container. This is the default."
+    echo ""
+    echo "    clean [<cargo args>]]"
+    echo "        Remove the Cloud Hypervisor artifacts."
+    echo ""
+    echo "    help"
+    echo "        Display this help message."
+    echo ""
+}
+
+cmd_build() {
+    build="debug"
+
+    while [ $# -gt 0 ]; do
+	case "$1" in
+            "-h"|"--help")  { cmd_help; exit 1;     } ;;
+            "--debug")      { build="debug";      } ;;
+            "--release")    { build="release";    } ;;
+            "--")           { shift; break;         } ;;
+            *)
+		die "Unknown build argument: $1. Please use --help for help."
+		;;
+	esac
+	shift
+    done
+
+    cargo_args=("$@")
+    [ $build = "release" ] && cargo_args+=("--release")
+
+    $DOCKER_RUNTIME run \
+	   -ti \
+	   --workdir "$CTR_CLH_ROOT_DIR" \
+	   --rm \
+	   --volume /dev:/dev \
+	   --volume "$CLH_ROOT_DIR:$CTR_CLH_ROOT_DIR" \
+	   "$CTR_IMAGE" \
+	   cargo build \
+	         --target-dir "$CTR_CLH_CARGO_TARGET" \
+	         "${cargo_args[@]}"
+
+    ret=$?
+
+    # If `cargo build` was successful, let's copy the binaries to a more
+    # accessible location.
+    [ $ret -eq 0 ] && {
+        cargo_bin_dir="$CLH_CARGO_TARGET/$build"
+        say "Binaries placed under $cargo_bin_dir"
+    }
+}
+
+cmd_clean() {
+    cargo_args=("$@")
+
+    $DOCKER_RUNTIME run \
+	   -ti \
+	   --workdir "$CTR_CLH_ROOT_DIR" \
+	   --rm \
+	   --volume "$CLH_ROOT_DIR:$CTR_CLH_ROOT_DIR" \
+	   "$CTR_IMAGE" \
+	   cargo clean \
+	         --target-dir "$CTR_CLH_CARGO_TARGET" \
+	         "${cargo_args[@]}"
+    }
+
+cmd_tests() {
+    unit=false
+    cargo=false
+    integration=false
+
+    while [ $# -gt 0 ]; do
+	case "$1" in
+            "-h"|"--help")    { cmd_help; exit 1;     } ;;
+            "--unit")         { unit=true;      } ;;
+            "--cargo")        { cargo=true;    } ;;
+	    "--integration")  { integration=true;    } ;;
+	    "--all")          { cargo=true; unit=true; integration=true;  } ;;
+            "--")             { shift; break;         } ;;
+            *)
+		die "Unknown tests argument: $1. Please use --help for help."
+		;;
+	esac
+	shift
+    done
+
+    if [ "$unit" = true ] ;  then
+	say "Running unit tests..."
+	$DOCKER_RUNTIME run \
+	       -ti \
+	       --workdir "$CTR_CLH_ROOT_DIR" \
+	       --rm \
+	       --privileged \
+	       --volume /dev:/dev \
+	       --volume "$CLH_ROOT_DIR:$CTR_CLH_ROOT_DIR" \
+	       "$CTR_IMAGE" \
+	       ./scripts/run_unit_tests.sh
+    fi
+
+    if [ "$cargo" = true ] ;  then
+	say "Running cargo tests..."
+	$DOCKER_RUNTIME run \
+	       -ti \
+	       --workdir "$CTR_CLH_ROOT_DIR" \
+	       --rm \
+	       --privileged \
```

Comment:
Is this needed?

Reply:
The arch unit tests need this, yes.

Reply:
Sorry, my misunderstanding: cargo test don't need to be privileged. Removed now.